### ISSUE TITLE
S3 path as explained in docs is wrong

### ISSUE THREAD

**Issue by cosimtaiuz** [aws-amplify/amplify-js#7447]
The docs say [here](https://docs.amplify.aws/lib/storage/configureaccess/q/platform/js):

> Protected: Readable by all users, but writable only by the creating user. Files are stored under protected/{user_identity_id}/ where the user_identity_id corresponds to the unique Amazon Cognito Identity ID for that user.

However when I look at files uploaded as "protected", the path for the user which uploaded them is not like that. The ID of the user doesn't match what I see there and there's also a prefix of the geographical area where the server is located, not just `{user_identity_id}`.

In any case, how can I retrieve the ID which is used to create the S3 protected folder for that user? Printing out the user info I get from `Auth.currentAuthenticatedUser()` doesn't show that ID at all.

**Comment by harrysolovay** [comment]
Odd. That `putStorageObject` helper that you point to is never actually used within the codebase. As far as the `Storage.put` method is concerned:

1. `put` immediately triggers the `Storage`-private [`_ensureCredentials`](https://github.com/aws-amplify/amplify-js/blob/c21e58aec0723ee0ee050682376ea4b95fbfe2ca/packages/storage/src/providers/AWSS3Provider.ts#L227), and returns a promise rejection if credential existence cannot be ensured.
2. a new `AWSS3ProviderUpload` is instantiated ([instantiation here](https://github.com/aws-amplify/amplify-js/blob/c21e58aec0723ee0ee050682376ea4b95fbfe2ca/packages/storage/src/providers/AWSS3Provider.ts#L294))
3. In the `AWSS3ProviderUpload`, we call a private `_createNewS3Client` method, which [re-gathers](https://github.com/aws-amplify/amplify-js/blob/c21e58aec0723ee0ee050682376ea4b95fbfe2ca/packages/storage/src/providers/AWSS3ProviderManagedUpload.ts#L334) the credentials and [passes them](https://github.com/aws-amplify/amplify-js/blob/c21e58aec0723ee0ee050682376ea4b95fbfe2ca/packages/storage/src/providers/AWSS3ProviderManagedUpload.ts#L347) into the S3 client constructor.

In summary: the underlaying client will contain different credentials for different users. Therefore, access to uploads should be scoped down per-user.

**Comment by cosimtaiuz** [comment]
@jamesonwilliams thanks for your reply; however when I visit the s3 bucket created by amplify, the path is not constructed neither with `'CognitoIdentityId-' + identityPoolId` + `/protected`;
instead, it's `the-region-name:some ID I can't find anywhere and which doesn't correspond to the user who created those files"/protected`

Hence I don't know how I could construct that path so that I can read those files from a lambda. I hope this makes more sense, if not do let me know and I can try to explain with some screenshots

**Comment by jamesonwilliams** [comment]
@cosimtaiuz We reviewed this internally, and I had misread the code. Apologies.

This line is looking up a saved identity ID from JavaScript's _local storage_:
```typescript
identityId = this._storage.getItem('CognitoIdentityId-' + identityPoolId);
```

A Cognito identity ID looks like this:
> us-east-1:5845b888-dbf5-4726-8947-6c5965206b10
```

It is a compound of a region and a unique ID (UUID). So, that's what you're seeing.

**Comment by cosimtaiuz** [comment]
@jamesonwilliams got you, that makes sense now. Thanks! How could I retrieve that ID from a lambda though?

**Comment by jamesonwilliams** [comment]
@cosimtaiuz This might be some good follow-up reading: https://serverless-stack.com/chapters/mapping-cognito-identity-id-and-user-pool-id.html

**Comment by cosimtaiuz** [comment]
Thanks @jamesonwilliams
